query-node/substrate-query-framework/index-builder/src/db/DatabaseManager.ts

```diff
@@ -0,0 +1,37 @@
+import { FindOneOptions, DeepPartial, EntityManager } from 'typeorm';
+
+import { DB } from '.';
+import { SubstrateEvent } from '..';
+import * as helper from './helper';
+
+/**
+ * Database access object based on typeorm. Use typeorm transactional entity manager to perform get/save/remove
+ * database operations.
+ *
+ * @constructor(event: QueryEvent, manager: EntityManager)
+ *
+ */
+export default class DatabaseManager implements DB {
+  // Transactional entity manager
+  private readonly _manager: EntityManager;
+
+  readonly event: SubstrateEvent;
+
+  constructor(event: SubstrateEvent, manager: EntityManager) {
```

Comment:
Is it right that we create a `DatabaseManager` per event? Shouldn't a manager class be more like a singleton?

Reply:
I have been thinking about this actually. Since we need to access `event` object via `db` in the mappings and let users perform only `CRUD` operations. I couldn't find any other solution. Maybe the name is misleading?

Reply:
I agree with @dzhelezov here, and `event` is not even used for anything internally in `DatabaseManager`. In the mapping, it currently just receivs the `DB` instance, but you could easily just change this, so you go from

`handleMemberRegistered(db: DB)` to `handleMemberRegistered(db: DB, event: SubstrateEvent)`

Is there any reason this would not work?